src/pages/edit.js

```diff
@@ -1,17 +1,43 @@
-import React from "react";
+import React, { useEffect } from 'react';
 import Layout from "../components/layout";
 import SEO from "../components/seo";
 import Editor from "../components/editor/Editor";
+import Auth0NavBar from '../components/auth0NavBar/auth0NavBar';
+import { Auth0Provider} from '@auth0/auth0-react';
 
 const edit = (props) => {
+  const query = window.location.search;
+  
+  if (query.includes("file=")) {
+    window.localStorage.setItem('previousPath', window.location.href);
+  }
+
+  useEffect(async () => {
+    const query = window.location.search;
+    if (query.includes("code=") && query.includes("state=")) {      
+      const href = window.localStorage.getItem('previousPath');
+      window.location.href = href;
+    }
```

Comment:
Since we cannot wild card the login redirect 

i.g.  `/edit?file=1234` -> login -> login success -> `/edit` 

You cannot do `/edit*` in the Auth0 configuration. So I save off the previous path they were at in the local storage. When the user gets redirected back to `/edit` and it is a success they are then redirected back to their original page they were on since we read it from local storage. It is kinda a hack but that is what Auth0 recommended.

OR you could do this within a callback redirect flow by passing data through the login request then reading that value when they get redirected but they have absolutely zero documentation on this. They just mention it but don't provide code snippets.

Reply:
- I pushed some changes to your branch as I was having issue with the edit page not loading the editor after logging in (tested with Brave and Chrome).  I ended up refactoring edit.js, moving Auth0 component to app's root component per their Gatsby example[1] for it to interact the url router.

- Protect the edit page with `export default withAuthenticationRequired(edit)`.  It takes care of the redirect after login.

[1] Auth0 Gatsby example:
https://github.com/auth0/auth0-react/blob/master/EXAMPLES.md#2-protecting-a-route-in-a-gatsby-app

### Is there a way to disable this screen when logging in for the first time?   

We can live with it for the time being.

<img width="436" alt="Screen Shot 2021-08-21 at 9 33 50 AM" src="https://user-images.githubusercontent.com/3805254/130314649-9b8b9b78-9090-434d-a619-a6e2f2381e36.png">